buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -71,6 +90,55 @@ public void apply(Project project) {
                         ImmutableList.of("-module-name", kotlinModuleName(project))));
   }
 
+  private static void setupApiInfomrationAnalysis(Project project, LibraryExtension android) {
+
+    String metalavaBinaryPath = System.getenv("METALAVA_BINARY_PATH");
+    AndroidSourceSet mainSourceSet = android.getSourceSets().getByName("main");
+    File outputFile = project.getRootProject().file(Paths.get(
+        project.getRootProject().getBuildDir().getPath(),
+        "apiinfo",
+        project.getPath().substring(1).replace(":", "_")));
+    File outputApiFile = new File(outputFile.getAbsolutePath() + "_api.txt");
+    String sourcePathArgument = mainSourceSet.getJava().getSrcDirs().stream()
+        .map(File::getAbsolutePath)
+        .collect(Collectors.joining(":"));
+    boolean doesSourcePathExist = false;
+    for(String sourcePath : sourcePathArgument.split(":")) {
+      if(new File(sourcePath).exists()) {
+        doesSourcePathExist = true;
```

Comment:
I think you can replace this for loop with `mainSourceSet.getJava().getSrcDirs().stream().anyMatch(File::exists)`